external/js/cothority/src/network/websocket.ts

```diff
@@ -39,18 +40,18 @@ export class WebSocketConnection implements IConnection {
         // by the fact that URL will not allow you to have an empty pathname,
         // which will always equal to "/" if there isn't any
         if (url.pathname.slice(-1) !== "/") {
-            url.pathname = url.pathname + "/";
+            url.set("pathname", url.pathname + "/");
         }
         if (url.username !== "" || url.password !== "") {
             throw new Error("addr contains authentication, which is not supported");
         }
-        if (url.search !== "" || url.hash !== "") {
+        if (url.query !== undefined || url.hash !== "") {
```

Comment:
The object is never `undefined`. The most precise test would be to check for the returned object size, which does works for both `string` and `Object`.

```suggestion
        if (Object.entries(url.query).length === 0 || url.hash !== "") {
```